app/src/main/java/com/stochanskyi/librariesdemo/presentaiton/feature/locationupdate/jobs/UpdateLocationWorker.kt

```diff
@@ -0,0 +1,77 @@
+package com.stochanskyi.librariesdemo.presentaiton.feature.locationupdate.jobs
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import androidx.work.CoroutineWorker
+import androidx.work.ForegroundInfo
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import com.stochanskyi.librariesdemo.R
+import com.stochanskyi.librariesdemo.app.appComponent
+import com.stochanskyi.librariesdemo.domain.features.location.GetCurrentLocationUseCase
+import kotlinx.coroutines.delay
+import javax.inject.Inject
+
+class UpdateLocationWorker(
+    appContext: Context,
+    params: WorkerParameters
+) : CoroutineWorker(appContext, params) {
+
+    companion object {
+        const val LAT_PROGRESS_KEY = "lat_progress_key"
+        const val LONG_PROGRESS_KEY = "long_progress_key"
+    }
+
+    @Inject
+    lateinit var getCurrentLocationUseCase: GetCurrentLocationUseCase
+
+    private val notificationManager by lazy {
+        applicationContext.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+    }
+
+    init {
+        appContext.appComponent().inject(this)
+    }
+
+    override suspend fun doWork(): Result {
+        while (!isStopped) {
+            val location = getCurrentLocationUseCase()
+            val notificationContent = applicationContext.getString(R.string.long_lat_format, location.long, location.lat)
+
+            setProgress(workDataOf(
+                LAT_PROGRESS_KEY to location.lat,
+                LONG_PROGRESS_KEY to location.long
+            ))
+
+            setForeground(createForegroundInfo(notificationContent))
+            delay(20000)
+        }
+
+        return Result.success()
+    }
+
+    private fun createForegroundInfo(content: String): ForegroundInfo {
+        return ForegroundInfo(12, createNotification(content))
```

Comment:
Notification ids should be moved out to a single place to avoid possible future collisions